crates/telio-crypto/src/lib.rs

```diff
@@ -42,6 +42,16 @@ pub struct SecretKey([u8; KEY_SIZE]);
 )]
 pub struct PublicKey(pub [u8; KEY_SIZE]);
 
+/// Canonical way to order keys
+pub fn key_order(lhs: &PublicKey, rhs: &PublicKey) -> Ordering {
+    lhs.cmp(rhs)
+}
```

Comment:
I fell that such an explicit function is better to indicate significance of this particular order. The fact that this is the 'natural' order for this type is just a coincidence - had the heartbeat algorithm used the biggest key, this would be the reverse order.

Reply:
In such case it probably doesn't need to be an order, as we don't need transitivity here, but it might be just any full antisymmetric relation, i.e. we just need to know which node to pick, so imho returning a boolean, or maybe this `Winning` / `Losing` enum would be clearer? I think also some more verbose comment would be helpful, maybe with some explanation, why we need this functions and the RFC number? I don't think that "canonical" helps a lot here.

Also, I feel like all of this logic with picking the right node doesn't really touch the NAT state reporting, so maybe it could be moved to a separate PR?

Reply:
Oh, there is even a proper name for this one: https://en.wikipedia.org/wiki/Tournament_(graph_theory)

Reply:
Nat state reporting is being done *only* by the winning node so I'm not sure how to move it to a different PR.

Reply:
Well, "NAT state reporting" depends on "winning and losing nodes", but we can have "losing and winning nodes" merged without "NAT state reporting" with no harm?

Reply:
I see what you mean. Do you think it has to be in a separate PR or would it be enough to extract this to a separate (first) commit in this pr?

Reply:
A separate commit sounds like a great improvement, but I think a separate PR would be even better (and relatively simple).

Reply:
I think it's better to keep this in the same pr. If I move the key_order related changes to a separate PR it will look strange since there will be no other place where it is used and I think such refactoring should be done only when you have a second call site that will use the newly extracted function.